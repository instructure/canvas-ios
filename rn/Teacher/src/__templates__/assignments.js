//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: '1376',
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
  anonymize_students: false,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment({ assignment_group_id: '1' })],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
