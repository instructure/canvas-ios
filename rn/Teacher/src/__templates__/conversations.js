//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const conversation: Template<Conversation> = template({
  id: '123456789',
  subject: 'learn to write code',
  workflow_state: 'read',
  last_message: 'this is the worst class ever',
  last_message_at: '2037-06-01T05:59:00Z',
  last_authored_message: 'this is the worst class ever',
  last_authored_message_at: '2037-06-01T05:59:00Z',
  participants: [
    {
      id: '1234',
      name: 'artorias of the abyss',
    },
    {
      id: '6789',
      name: 'solaire of astora',
    },
  ],
  avatar_url: 'http://www.fillmurray.com/200/300',
  context_name: 'Test Class',
  context_code: 'course_1',
  audience_contexts: {},
  audience: [],
  message_count: 2,
  private: false,
  properties: null,
  starred: false,
  subscribed: false,
  visible: true,
})

export const conversationMessage: Template<ConversationMessage> = template({
  id: '123456789',
  body: 'this is neat',
  author_id: '1234',
  created_at: '2037-06-01T05:59:00Z',
  attachments: [
    {
      id: '1',
      display_name: 'something.pdf',
      mime_class: 'pdf',
    },
  ],
  generated: false,
})
