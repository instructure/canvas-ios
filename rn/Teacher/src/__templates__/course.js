//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import { CourseModel } from '../canvas-api/model'
import template, { type Template } from '../utils/template'
import { section } from './section'

export const courseTerm: Template<CourseTerm> = template({
  id: '1',
  created_at: '2017-03-17T19:15:25Z',
  name: 'Default Term',
})

export const course: Template<Course> = template({
  id: '1',
  account_id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
  permissions: {
    viewAnalytics: true,
    viewAllGrades: true,
  },
  workflow_state: 'available',
  term: {
    id: '1',
    created_at: '2018-02-08T23:13:39Z',
    end_at: '2030-02-08T23:13:39Z',
    start_at: '2018-02-08T23:13:39Z',
    name: 'Default Term',
    workflow_state: 'available',
  },
  end_at: '2030-02-08T23:13:39Z',
})

export const courseModel: Template<CourseModel> = overrides =>
  Object.assign(new CourseModel(course()), overrides)

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})

export const courseSettings: Template<CourseSettings> = template({
  allow_student_forum_attachments: true,
})
