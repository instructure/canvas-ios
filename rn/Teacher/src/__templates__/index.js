//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
