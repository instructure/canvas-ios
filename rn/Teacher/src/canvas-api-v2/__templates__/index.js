// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
