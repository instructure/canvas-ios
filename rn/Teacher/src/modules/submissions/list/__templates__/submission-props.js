//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
