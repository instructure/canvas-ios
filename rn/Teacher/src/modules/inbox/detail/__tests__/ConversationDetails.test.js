//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import { shallow } from 'enzyme'
import * as React from 'react'
import {
  ActionSheetIOS,
  Alert,
} from 'react-native'
import * as template from '../../../../__templates__'
import {
  ConversationDetails,
  handleRefresh,
  mapStateToProps,
  shouldRefresh,
} from '../ConversationDetails'

jest
  .mock('react-native/Libraries/Alert/Alert', () => ({ alert: jest.fn() }))
  .mock('react-native/Libraries/ActionSheetIOS/ActionSheetIOS', () => ({
    showActionSheetWithOptions: jest.fn(),
  }))

describe('ConversationDetails', () => {
  let props
  beforeEach(() => {
    jest.clearAllMocks()
    props = {
      conversation: template.conversation({ id: '1' }),
      conversationID: '1',
      messages: [],
      navigator: template.navigator(),
      enrollments: [template.enrollment()],
      refreshEnrollments: jest.fn(),
      refreshInboxAll: jest.fn(),
      refreshInboxUnread: jest.fn(),
      refreshInboxStarred: jest.fn(),
      refreshInboxSent: jest.fn(),
      refreshInboxArchived: jest.fn(),
      updateInboxSelectedScope: jest.fn(),
      refreshConversationDetails: jest.fn(),
      starConversation: jest.fn(),
      unstarConversation: jest.fn(),
      deleteConversation: jest.fn(),
      deleteConversationMessage: jest.fn(),
      markAsRead: jest.fn(),
    }
  })

  it('renders correctly', () => {
    const tree = shallow(<ConversationDetails {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with no conversation', () => {
    props.conversation = null
    const tree = shallow(<ConversationDetails {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with some messages', () => {
    props.messages = [template.conversationMessage()]
    const tree = shallow(<ConversationDetails {...props} />)
    expect(tree).toMatchSnapshot()
    expect(tree.find('FlatList').dive()).toMatchSnapshot()
  })

  it('renders with alternate data', () => {
    props.conversation = template.conversation({ id: '1', starred: true, subject: null })
    const tree = shallow(<ConversationDetails {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('doesnt render messages that are pendingDelete', () => {
    props.messages = [template.conversationMessage({ pendingDelete: true })]
    const tree = shallow(<ConversationDetails {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('shows options action sheet', () => {
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalledWith(
      {
        options: [ 'Reply', 'Reply All', 'Forward', 'Delete', 'Cancel' ],
        destructiveButtonIndex: 3,
        cancelButtonIndex: 4,
      },
      expect.any(Function),
    )
  })

  it('replies only to author when Reply is chosen', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(0))
    const conversation = props.conversation = template.conversation({ audience: [ '1234', '6789' ] })
    props.conversationID = '1'
    props.messages = [
      template.conversationMessage({ id: '1', author_id: '1234' }),
    ]
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/conversations/1/add_message',
      { modal: true },
      {
        recipients: conversation.participants.slice(0, 1),
        contextName: conversation.context_name,
        contextCode: conversation.context_code,
        subject: conversation.subject,
        canSelectCourse: false,
        canEditSubject: false,
        navBarTitle: 'Reply',
      }
    )
  })

  it('replies to all when Reply is chosen on your own message', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(0))
    const conversation = props.conversation = template.conversation({ audience: [ '1234', '6789' ] })
    props.conversationID = '1'
    props.messages = [
      template.conversationMessage({ id: '1', author_id: '1' }),
    ]
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action('1')
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/conversations/1/add_message',
      { modal: true },
      expect.objectContaining({
        recipients: conversation.participants,
      })
    )
  })

  it('replies to all when Reply All is chosen', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(1))
    const conversation = props.conversation = template.conversation({ audience: [ '1234', '6789' ] })
    props.conversationID = '1'
    props.messages = [
      template.conversationMessage({ id: '1', author_id: '1' }),
    ]
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.show).toHaveBeenCalledWith(
      '/conversations/1/add_message',
      { modal: true },
      expect.objectContaining({
        recipients: conversation.participants,
      })
    )
  })

  it('calls deleteConversation from options', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.deleteConversation = jest.fn()
    props.conversationID = '1'
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.deleteConversation).toHaveBeenCalledWith('1')
  })

  it('should toggle starred', () => {
    props.conversation = template.conversation({ starred: false })
    const tree = shallow(<ConversationDetails {...props} />)
    shallow(tree.find('FlatList').prop('ListHeaderComponent'))
      .find('[testID="inbox.detail.not-starred"]').simulate('Press')
    expect(props.starConversation).toHaveBeenCalled()
    tree.setProps({ conversation: { ...props.conversation, starred: true } })
    shallow(tree.find('FlatList').prop('ListHeaderComponent'))
      .find('[testID="inbox.detail.starred"]').simulate('Press')
    expect(props.unstarConversation).toHaveBeenCalled()
  })

  it('passess all messages as included messages when forwarding the conversation', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(2))
    props.conversationID = '1'
    props.messages = [
      template.conversationMessage({ id: '1' }),
      template.conversationMessage({ id: '2' }),
    ]
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.show).toHaveBeenCalledWith('/conversations/1/add_message', { modal: true }, {
      contextName: props.conversation && props.conversation.context_name,
      contextCode: props.conversation && props.conversation.context_code,
      subject: `Fw: ${props.conversation && props.conversation.subject || ''}`,
      showCourseSelect: false,
      canEditSubject: false,
      navBarTitle: 'Forward',
      requireMessageBody: false,
      includedMessages: props.messages,
    })
  })

  it('only passes one message when forwarding a single message', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(2))
    props.conversationID = '1'
    props.messages = [template.conversationMessage({ id: '3' })]
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action('3')
    expect(props.navigator.show).toHaveBeenCalledWith('/conversations/1/add_message', { modal: true }, {
      contextName: props.conversation && props.conversation.context_name,
      contextCode: props.conversation && props.conversation.context_code,
      subject: `Fw: ${props.conversation && props.conversation.subject || ''}`,
      showCourseSelect: false,
      canEditSubject: false,
      navBarTitle: 'Forward',
      requireMessageBody: false,
      includedMessages: [props.messages[0]],
    })
  })

  it('does not allow forwarding of messages with no context_code', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(2))
    props.conversationID = '1'
    props.messages = [template.conversationMessage({ id: '3' })]
    props.conversation = template.conversation({ id: '1', context_code: undefined })
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action('3')
    expect(Alert.alert).toHaveBeenCalled()
    expect(props.navigator.show).not.toHaveBeenCalled()
  })

  it('calls options with an id', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.conversationID = '1'
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action('2')
    expect(props.deleteConversationMessage).toHaveBeenCalledWith('1', '2')
  })

  it('calls navigator.pop when there is no conversation', () => {
    const tree = shallow(<ConversationDetails {...props} />)
    tree.setProps({ conversation: undefined })
    expect(props.navigator.pop).toHaveBeenCalled()
  })

  it('calls navigator.dismiss when there is no conversation and is modal', () => {
    props.navigator.isModal = true
    const tree = shallow(<ConversationDetails {...props} />)
    tree.setProps({ conversation: undefined })
    expect(props.navigator.dismiss).toHaveBeenCalled()
  })

  it('calls pop after delete finishes', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.deleteConversation = jest.fn(() => {
      tree.setProps({ pending: 1 })
      tree.setProps({ pending: 0, conversation: null })
    })
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.pop).toHaveBeenCalled()
  })

  it('calls dismiss after delete finishes if modal', () => {
    // $FlowFixMe
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((config, callback) => callback(config.options.length - 2))
    props.navigator.isModal = true
    props.deleteConversation = jest.fn(() => {
      tree.setProps({ pending: 1 })
      tree.setProps({ pending: 0, conversation: null })
    })
    const tree = shallow(<ConversationDetails {...props} />)
    tree.find('Screen').prop('rightBarButtons')[0].action()
    expect(props.navigator.dismiss).toHaveBeenCalled()
  })

  it('calls markAsRead on componentDidMount when the conversation has workflow state of unread', () => {
    const myProps = {
      ...props,
      conversation: template.conversation({ workflow_state: 'unread' }),
    }
    shallow(<ConversationDetails {...myProps} />)
    expect(props.markAsRead).toHaveBeenCalledWith('1')
  })

  it('does not call markAsRead on componentDidMount when the conversation has workflow state of archived', () => {
    const myProps = {
      ...props,
      conversation: template.conversation({ workflow_state: 'archived' }),
    }
    shallow(<ConversationDetails {...myProps} />)
    expect(props.markAsRead).not.toHaveBeenCalledWith('1')
  })

  describe('mapStateToProps', () => {
    const c1 = template.conversation({
      id: '1',
    })
    const appState = template.appState({
      entities: {
        courses: {
          '1': {
            enrollments: {
              refs: ['1'],
            },
          },
        },
        enrollments: {
          '1': template.enrollment(),
        },
      },
      inbox: {
        conversations: {
          '1': { data: c1 },
        },
        selectedScope: 'all',
        all: { refs: [c1.id] },
        unread: { refs: [] },
        starred: { refs: [] },
      },
    })

    it('maps correctly', () => {
      const results = mapStateToProps(appState, { conversationID: c1.id })
      expect(results).toMatchObject({
        conversation: c1,
      })
    })
  })

  describe('handleRefresh', () => {
    const props = {
      refreshConversationDetails: jest.fn(),
      starConversation: jest.fn(),
      unstarConversation: jest.fn(),
      markAsRead: jest.fn(),
      conversationID: '1',
      conversation: null,
      messages: [],
      navigator: template.navigator(),
    }

    it('refreshes', () => {
      handleRefresh(props)
      expect(props.refreshConversationDetails).toHaveBeenCalledWith('1')
    })
  })

  describe('shouldRefresh', () => {
    it('should refresh', () => {
      const conversation = template.conversation({ messages: null })
      const props = {
        conversation,
        conversationID: conversation.id,
        messages: [],
      }

      expect(shouldRefresh(props)).toEqual(true)
    })
  })
})
