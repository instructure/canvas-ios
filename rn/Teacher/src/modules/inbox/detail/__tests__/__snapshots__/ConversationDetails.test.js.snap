// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationDetails doesnt render messages that are pendingDelete 1`] = `
<Screen
  customPageViewPath="/conversations"
  drawUnderNavBar={true}
  navBarStyle="global"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Conversation options",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/kabob.png",
        },
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  showDismissButton={true}
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
              "paddingBottom": 8,
            }
          }
        >
          <Heading1>
            learn to write code
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Un-starred"
            accessibilityTraits="button"
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.not-starred"
          >
            <Image
              source={
                Object {
                  "uri": "starLine",
                }
              }
              style={
                Object {
                  "height": 24,
                  "tintColor": "#008EE2",
                  "width": 24,
                }
              }
            />
          </TouchableOpacity>
        </View>
      }
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
  </View>
</Screen>
`;

exports[`ConversationDetails renders correctly 1`] = `
<Screen
  customPageViewPath="/conversations"
  drawUnderNavBar={true}
  navBarStyle="global"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Conversation options",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/kabob.png",
        },
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  showDismissButton={true}
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
              "paddingBottom": 8,
            }
          }
        >
          <Heading1>
            learn to write code
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Un-starred"
            accessibilityTraits="button"
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.not-starred"
          >
            <Image
              source={
                Object {
                  "uri": "starLine",
                }
              }
              style={
                Object {
                  "height": 24,
                  "tintColor": "#008EE2",
                  "width": 24,
                }
              }
            />
          </TouchableOpacity>
        </View>
      }
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
  </View>
</Screen>
`;

exports[`ConversationDetails renders correctly with no conversation 1`] = `
<Screen
  customPageViewPath="/conversations"
  drawUnderNavBar={true}
  navBarStyle="global"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Conversation options",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/kabob.png",
        },
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  showDismissButton={true}
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={<View />}
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
  </View>
</Screen>
`;

exports[`ConversationDetails renders correctly with some messages 1`] = `
<Screen
  customPageViewPath="/conversations"
  drawUnderNavBar={true}
  navBarStyle="global"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Conversation options",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/kabob.png",
        },
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  showDismissButton={true}
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
              "paddingBottom": 8,
            }
          }
        >
          <Heading1>
            learn to write code
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Un-starred"
            accessibilityTraits="button"
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.not-starred"
          >
            <Image
              source={
                Object {
                  "uri": "starLine",
                }
              }
              style={
                Object {
                  "height": 24,
                  "tintColor": "#008EE2",
                  "width": 24,
                }
              }
            />
          </TouchableOpacity>
        </View>
      }
      data={
        Array [
          Object {
            "attachments": Array [
              Object {
                "display_name": "something.pdf",
                "id": "1",
                "mime_class": "pdf",
              },
            ],
            "author_id": "1234",
            "body": "this is neat",
            "created_at": "2037-06-01T05:59:00Z",
            "generated": false,
            "id": "123456789",
          },
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
  </View>
</Screen>
`;

exports[`ConversationDetails renders correctly with some messages 2`] = `
<list>
  <item
    key="123456789"
  >
    <ConversationMessageRow
      conversation={
        Object {
          "audience": Array [],
          "audience_contexts": Object {},
          "avatar_url": "http://www.fillmurray.com/200/300",
          "context_code": "course_1",
          "context_name": "Test Class",
          "id": "1",
          "last_authored_message": "this is the worst class ever",
          "last_authored_message_at": "2037-06-01T05:59:00Z",
          "last_message": "this is the worst class ever",
          "last_message_at": "2037-06-01T05:59:00Z",
          "message_count": 2,
          "participants": Array [
            Object {
              "id": "1234",
              "name": "artorias of the abyss",
            },
            Object {
              "id": "6789",
              "name": "solaire of astora",
            },
          ],
          "private": false,
          "properties": null,
          "starred": false,
          "subject": "learn to write code",
          "subscribed": false,
          "visible": true,
          "workflow_state": "read",
        }
      }
      firstMessage={true}
      message={
        Object {
          "attachments": Array [
            Object {
              "display_name": "something.pdf",
              "id": "1",
              "mime_class": "pdf",
            },
          ],
          "author_id": "1234",
          "body": "this is neat",
          "created_at": "2037-06-01T05:59:00Z",
          "generated": false,
          "id": "123456789",
        }
      }
      navigator={
        Object {
          "dismiss": [MockFunction],
          "dismissAllModals": [MockFunction],
          "isModal": false,
          "pop": [MockFunction],
          "replace": [MockFunction],
          "show": [MockFunction],
          "showWebView": [MockFunction],
          "traitCollection": [MockFunction],
        }
      }
      onReply={[Function]}
      showOptionsActionSheet={[Function]}
    />
  </item>
</list>
`;

exports[`ConversationDetails renders with alternate data 1`] = `
<Screen
  customPageViewPath="/conversations"
  drawUnderNavBar={true}
  navBarStyle="global"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Conversation options",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/kabob.png",
        },
        "testID": "inbox.detail.options.button",
      },
    ]
  }
  showDismissButton={true}
  title="Message Details"
>
  <View
    style={
      Object {
        "backgroundColor": "#F5F5F5",
        "flex": 1,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderTopColor": "#C7CDD1",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 16,
              "paddingBottom": 8,
            }
          }
        >
          <Heading1>
            No Subject
          </Heading1>
          <TouchableOpacity
            accessibilityLabel="Starred"
            accessibilityTraits="button"
            focusedOpacity={0.7}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onPress={[Function]}
            testID="inbox.detail.starred"
          >
            <Image
              source={
                Object {
                  "uri": "starSolid",
                }
              }
              style={
                Object {
                  "height": 24,
                  "tintColor": "#008EE2",
                  "width": 24,
                }
              }
            />
          </TouchableOpacity>
        </View>
      }
      data={Array []}
      keyExtractor={[Function]}
      renderItem={[Function]}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
        }
      }
    />
  </View>
</Screen>
`;
